### PR TITLE
CALS-5462 Reset or Clear Function for Search Criteria

### PR DESCRIPTION
**Pull request by sunilb0575**
#### Which User story does this relate to?
- https://osi-cwds.atlassian.net/browse/CALS-5462
-https://osi-cwds.atlassian.net/browse/CALS-5688

#### Brief feature description
-Add reset button to clear search criteria and search results
-change label 'Facility license/Approval #'  to 'Facility ID/Approval #' in list view
 -              **search.js**   **search_input.js**    **search_list**
- Refactoring search and facility profile,
- Changing child components from class to functional component

#### Does this PR block/effect other stories?
- No

#### Do any commands needs to be run in order to run the project with this PR?
- No

#### Code coverage percent diff as a result of this PR?
- statements: 83, branches: 70, functions: 79, lines: 83

### CODE REVIEW

File: app/javascript/facility/complaints.jsx [hunk 1]
```diff
@@ -4,49 +4,43 @@ import {checkforDateOrNa} from 'search/common/commonUtils'
 
 const title = 'Complaint History'
```
Comment (Nasir2712) [inline app/javascript/facility/complaints.jsx]: unnecessary variable. can you just use the string in the h3 tag

File: app/javascript/search/search_Data.js [hunk 2]
```diff
@@ -3,89 +3,72 @@ import PropTypes from 'prop-types'
...
+  sizeValue,
+  pageNumber,
+  searchApiCall,
+  handleToggle,
+  changePage
+}) => (
+  <div className='search-toggle col-xs-12 col-sm-12 col-md-12 col-lg-12'>
+    <div className='col-xs-12 col-sm-12 col-md-12 col-lg-12'>
+    </div>
+    <div className='search_details col-xs-12 col-sm-9 col-md-9 col-lg-9'>
```
Comment (Nasir2712) [inline app/javascript/search/search_Data.js]: can you give 10 width and 2 for the search button div?

Comment (sunilb0575) [reply app/javascript/search/search_Data.js]: @Nasir2712  changed width for both <div> field_input and field_search, in search_input.

File: app/javascript/facility/children.jsx [hunk 3]
```diff
@@ -2,57 +2,47 @@ import React from 'react'
...
-                <th> DATE OF BIRTH </th>
-                <th> DATE OF PLACEMENT </th>
-                <th> ASSIGNED WORKER </th>
-                <th> COUNTY OF ORIGIN </th>
+const Children = ({
+  facilityData
+}) => (
+  <div className='facility-children col-xs-12 col-sm-12 col-md-12 col-lg-12'>
+    <div className='facility-children-block'>
+      <div className='children-title'> <h3> {'Children currently placed in facility'} </h3> </div>
```
Comment (Nasir2712) [inline app/javascript/facility/children.jsx]: remove braces around text.

File: app/javascript/facility/complaints.jsx [hunk 4]
```diff
@@ -2,51 +2,43 @@ import React from 'react'
...
-                <th> CONTROL NUMBER </th>
-                <th> PRIORITY </th>
-                <th> STATUS </th>
-                <th> APPROVAL DATE </th>
+const Complaints = ({
+  facilityData
+}) => (
+  <div className='facility-children col-xs-12 col-sm-12 col-md-12 col-lg-12'>
+    <div className='facility-children-block'>
+      <div className='children-title'> <h3> {'Complaint History'} </h3> </div>
```
Comment (Nasir2712) [inline app/javascript/facility/complaints.jsx]: same here

File: spec/features/show_facility_test_spec.rb [hunk 5]
```diff
@@ -77,20 +77,20 @@
...
+    visit search_index_path
+    find_button('search').click
+    expect(find_field('dropdownFacilities').value).to eq '10'
+    expect(page).to have_css(:span, text: '1')
+    expect(page).to have_css(:span, text: '330')
+    find(:css, '#next_button').click
+    expect(page).to have_text('2')
+    select 20, :from => "dropdownFacilities"
+    expect(find_field('dropdownFacilities').value).to eq '20'
+    expect(page).to have_text('Zamora, Lupe')
```
Comment (ratneshraval) [inline spec/features/show_facility_test_spec.rb]: Remove the hard text verification. Just check that facilities exists by checking length. So if sort or data changes, you don't have this fail.

File: test/javascript/search/search_tests.js [hunk 6]
```diff
@@ -94,4 +96,14 @@ describe('Verify Search component', function () {
     countyDropDownChange.simulate('change', {target: {options: {'19': {id: '19', value: 'Los Angeles'}, selectedIndex: 19}}})
     expect(handleInputChangeSpy).toHaveBeenCalledWith('countyValue', 'Los Angeles')
   })
+  it('verify clicking reset button calls resetForm method', () => {
+    let resetForm = searchComp.find('#reset')
+    resetForm.simulate('click')
+    expect(resetFormSpy).toHaveBeenCalled()
```
Comment (ratneshraval) [inline test/javascript/search/search_tests.js]: Please validate new state by using
```
 expect(searchComp.state('inputData')).toEqual({})
 expect(searchComp.state('searchResults')).toEqual(undefined)
```

File: app/javascript/search/search.js [hunk 7]
```diff
@@ -37,9 +38,15 @@ export default class Search extends React.Component {
     })
   }
 
-  removeCriteria (value) {
-    this.handleInputChange(value, '')
-    this.searchApiCall(0, this.state.sizeValue)
+  handleOnSubmit (event) {
+    event.preventDefault()
+    this.searchApiCall(0, 10)
```
Comment (ratneshraval) [inline app/javascript/search/search.js]: This looks like a mistake. Should you use `this.searchApiCall(0, this.state.sizeValue)`?

File: test/javascript/search/searchInput_tests.js [hunk 8]
```diff
@@ -65,9 +65,9 @@ describe('Verify search input component', function () {
     expect(spyHandleInputChange).toHaveBeenCalledWith('facilityAddressValue', '36 Sequoia Dr, Aliso Viejo, CA 92656')
   })
 
-  it('verify clicking search button calls searchApiCall method', () => {
-    let searchFacility = searchInputComp.find('.btn-primary')
+  it('verify clicking search button calls handleOnSubmit method', () => {
+    let searchFacility = searchInputComp.find('#search')
     searchFacility.simulate('submit')
-    expect(spySearchApiCall).toHaveBeenCalledWith(0, 10)
+    expect(spyHandleOnSubmit).toHaveBeenCalled()
```
Comment (ratneshraval) [inline test/javascript/search/searchInput_tests.js]: Maybe add a further line to verify that searchApi is called with proper size value

File: app/javascript/search/search.js (hunk 7 above)
Comment (sunilb0575) [reply app/javascript/search/search.js]: Passing 10 for initial load, if I pass this.state.sizeValue, value in the dropdown(for changing number of results) is not reverting back to the initial load.

Comment (ratneshraval) [reply app/javascript/search/search.js]: I think that should be the goal, the value you set, should persist during next searches while you're on the page.

**Pull request merged**